### PR TITLE
fix #902, delete kafka topic on close

### LINKED ISSUE #902: KafkaSource should provide an option to either delete a topic or to auto delete a topic on close
See #892

### PR DESCRIPTION
**Pull request by manuzhang**

### CODE REVIEW

File: external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/KafkaUtil.scala [hunk 1]
```diff
@@ -87,6 +87,17 @@ object KafkaUtil {
...
   }
 
+  def deleteTopic(zkClient: ZkClient, topic: String): Unit = {
+    try {
+      AdminUtils.deleteTopic(zkClient, topic)
+    } catch {
+      case e: Exception =>
+        LOG.error(e.getMessage)
+    } finally {
+      zkClient.close()
```
Comment (stanleyxu2005) [inline external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/KafkaUtil.scala]: Shouldn't `zkClient` be closed in `KafkaStorage.scala`?

Comment (manuzhang) [inline external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/KafkaUtil.scala]: Good point but I don't want `KafkaStorage` to worry about the zookeeper connection, whose creation and close should be contained in this method. Now I passed in a connect function so there will be no `ZkClient` objects in `KafkaStorage`. I also applied this to other methods that interact with zookeeper. The reason I don't just create a `ZkClient` here is to make the method testable.

**Comment by clockfly** [comment]
+1

**Comment by clockfly** [comment]
+1

**Pull request merged**